bitswap/network/http_multiaddr_test.go

```diff
@@ -0,0 +1,183 @@
+package network
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multiaddr"
+)
+
+func TestExtractHTTPAddress(t *testing.T) {
+	tests := []struct {
+		name      string
+		maStr     string
+		want      *url.URL
+		sni       string
+		expectErr bool
+	}{
+		{
+			name:  "Valid HTTP multiaddress with DNS",
+			maStr: "/dns4/example.com/tcp/8080/http",
+			want: &url.URL{
+				Scheme: "http",
+				Host:   "example.com:8080",
+			},
+			expectErr: true,
+		},
+		{
+			name:  "Valid HTTPS multiaddress with DNS",
+			maStr: "/dns4/example.com/tcp/443/https",
+			want: &url.URL{
+				Scheme: "https",
+				Host:   "example.com:443",
+			},
+			expectErr: false,
+		},
```

Comment:
nit: include regression test for `/dns6` and `/dns`

```suggestion
			name:  "Valid HTTPS multiaddress with DNS4",
			maStr: "/dns4/example.com/tcp/443/https",
			want: &url.URL{
				Scheme: "https",
				Host:   "example.com:443",
			},
			expectErr: false,
		},
		{
		  name:  "Valid HTTPS multiaddress with DNS6",
			maStr: "/dns6/example.com/tcp/443/https",
			want: &url.URL{
				Scheme: "https",
				Host:   "example.com:443",
			},
			expectErr: false,
		},
		{
		  name:  "Valid HTTPS multiaddress with DNS",
			maStr: "/dns/example.com/tcp/443/https",
			want: &url.URL{
				Scheme: "https",
				Host:   "example.com:443",
			},
			expectErr: false,
		},		
```

Reply:
3ea55a6ddc163561c15b2e57e5565f1eb6e17e06